### PR TITLE
feat: handle empty name property

### PR DESCRIPTION
**Pull request by eranshmil**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[x] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
When name property is undefined, the error `Cannot read property 'split' of undefined` is raised from the @angular/devkit package.

## What is the new behavior?
The following has been added to store, state and actions schematics:
- https://github.com/ngxs/schematics/pull/22/commits/519074b43a021e009cb7e1d98733399293229022 https://github.com/ngxs/schematics/pull/22/commits/47ddfeb1f38fb62c6212b1c1ae7d40f19ca63164 If the name is empty - a prompt is shown to enter the required name: `? What name would you like to use for the store?`.
- https://github.com/ngxs/schematics/pull/22/commits/5c3d96b2b2a5ed37379faef9fe6ff80923a4d922 If the trimmed name is empty - an error is shown: `Invalid options, "name" is required.`.

Also, the following has been changed (**Sorry for doing this in the same PR, there were not much changes when commiting them, if someone wants I'll seperate them another PRs**):
- https://github.com/ngxs/schematics/commit/f102638c292b7be5b482c8f046e6473f6968f84a The if inside factoryException is never used, the prop is only a string that represents the property in the options.
Also, this code is applied to all the schematics, while only 3 of them have the name property (store, state & actions).
- https://github.com/ngxs/schematics/commit/dffb208a9c307f39ea04cd1eef61d86276d64ed4 Fix typo in `tranform`.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by eranshmil** [comment]
@kyusupov33 Please also take a look.

File: src/factories/store/store.factory.ts [hunk 1]
```diff
@@ -1,8 +1,14 @@
...
+
+import { FACTORIES, isEmpty } from '../../utils';
 import { factoryLoader } from '../../utils/factory-loader';
 
+import { StoreSchema } from './store.schema';
+
 export function store(options: StoreSchema): Rule {
+  if (isEmpty(options.name)) {
+    throw new SchematicsException('Invalid options, "name" is required.');
+  }
```
Comment (splincode) [inline src/factories/store/store.factory.ts]:
```ts
if (isEmpty(options.name)) {
     throw new SchematicsException('Invalid options, "name" is required.');
   }
```

Current code used in two places (action.factory.ts, store.factory.ts), I think this code need moved to factory-loader.ts

File: src/utils/factory-loader.ts [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+import { FACTORIES } from './common/factories.enum';
 import { StarterKitSchema } from '../factories/starter-kit/starter-kit.schema';
-import { mergeWith, Rule } from '@angular-devkit/schematics';
+
+import { transform } from './transform-options';
 import { generate } from './generate-factory';
-import { FACTORIES } from './common/factories.enum';
 
 export function factoryLoader<T>(options: T | any, factory: FACTORIES): Rule {
+  if (typeof options.name !== 'undefined' && !options.name.trim().length) {
```
Comment (splincode) [inline src/utils/factory-loader.ts]: conditions should be easy to read

```ts
if (Utils.valueIsEmpty(option.name)) {
..
```

File: src/utils/common/properties.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export function isEmpty(value: string | undefined): boolean {
+  return typeof value !== 'undefined' && !value.trim().length;
+}
```
Comment (deebloo) [inline src/utils/common/properties.ts]: isEmpty only checks if a string is defined right? if so maybe we should name it something more specific

**Pull request merged**